packages/graphql-client/src/graphql-client/graphql-client.ts

```diff
@@ -165,3 +184,276 @@ function generateRequest(
     }
   };
 }
+
+async function* getStreamBodyIterator(
+  response: Response,
+): AsyncIterableIterator<string> {
+  // Support node-fetch format
+  if ((response.body as any)![Symbol.asyncIterator]) {
+    for await (const chunk of response.body! as any) {
+      yield (chunk as Buffer).toString();
+    }
+  } else {
+    const reader = response.body!.getReader();
+    const decoder = new TextDecoder();
+
+    let readResult: ReadableStreamReadResult<DataChunk>;
+    try {
+      while (!(readResult = await reader.read()).done) {
+        yield decoder.decode(readResult.value);
+      }
+    } finally {
+      reader.cancel();
+    }
+  }
+}
+
+function readStreamChunk(
+  streamBodyIterator: AsyncIterableIterator<string>,
+  boundary: string,
+) {
+  return {
+    async *[Symbol.asyncIterator]() {
+      try {
+        let buffer = "";
+
+        for await (const textChunk of streamBodyIterator) {
+          buffer += textChunk;
+
+          if (buffer.indexOf(boundary) > -1) {
```

Comment:
If we get back a chunk that doesn't have a boundary, is that something we should throw on? Is it possible?